lib/src/main/java/org/openapitools/openapistylevalidator/custom/OpenAPIRule.java

```diff
@@ -0,0 +1,16 @@
+package org.openapitools.openapistylevalidator.custom;
+
+import org.eclipse.microprofile.openapi.models.OpenAPI;
+import org.openapitools.openapistylevalidator.ErrorAggregator;
+
+public abstract class OpenAPIRule {
+
+    protected ErrorAggregator errorAggregator;
```

Comment:
I see why you put the ErrorAggregator public... Maybe if we move some stuff around this will not be required... I will let you fix some stuff first and then I could help you to finish this PR.